api/v1beta1/disruption_schedule_types.go

```diff
@@ -0,0 +1,87 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package v1beta1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+func init() {
+	SchemeBuilder.Register(&DisruptionSchedule{}, &DisruptionScheduleList{})
+}
+
+//+kubebuilder:object:root=true
+
+// DisruptionSchedule is the Schema for the disruptions API
+// +kubebuilder:resource:shortName=disch
+// +kubebuilder:subresource:status
+type DisruptionSchedule struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              DisruptionScheduleSpec   `json:"spec,omitempty"`
+	Status            DisruptionScheduleStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// DisruptionScheduleList contains a list of DisruptionSchedule
+type DisruptionScheduleList struct {
```

Comment:
I thought it had to be defined for any new CRD?

Reply:
Based on some of the research I did. This `List` struct for the CRD is automatically created by Kubebuilder. We won't need to directly work with the `List` struct. It's mostly used by the client-go library under the hood to interact with the Kubernetes API server. It is especially useful for list/watch patterns in controllers. If we remove it, it might break these patterns.